### ISSUE TITLE
java.nio.file.FileSystemAlreadyExistsException: null with latest snapshot

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#11144]
This is from the build log of a Spring snapshot
```
[16:23:06] :	 [Step 3/5] java.nio.file.FileSystemAlreadyExistsException: null
[16:23:06] :	 [Step 3/5] 	at com.sun.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:113) ~[zipfs.jar:1.8.0_201]
[16:23:06] :	 [Step 3/5] 	at java.nio.file.FileSystems.newFileSystem(FileSystems.java:326) ~[na:1.8.0_201]
[16:23:06] :	 [Step 3/5] 	at java.nio.file.FileSystems.newFileSystem(FileSystems.java:276) ~[na:1.8.0_201]
[16:23:06] :	 [Step 3/5] 	at com.vaadin.flow.server.StaticFileServer.resourceIsDirectory(StaticFileServer.java:146) ~[flow-server-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
[16:23:06] :	 [Step 3/5] 	at com.vaadin.flow.server.StaticFileServer.isStaticResourceRequest(StaticFileServer.java:118) ~[flow-server-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
[16:23:06] :	 [Step 3/5] 	at com.vaadin.flow.server.VaadinServlet.serveStaticOrWebJarRequest(VaadinServlet.java:343) ~[flow-server-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
[16:23:06] :	 [Step 3/5] 	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:289) ~[flow-server-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
[16:23:06] :	 [Step 3/5] 	at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:109) ~[vaadin-spring-18.0-SNAPSHOT.jar:na]
[16:23:06] :	 [Step 3/5] 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) [tomcat-embed-core-9.0.45.jar:4.0.FR]
```

Seems like Flow is trying to create multiple file system instances for the same zip file which based on this code can never work: https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/jdk.zipfs/share/classes/jdk/nio/zipfs/ZipFileSystemProvider.java#L90-L121

Potentially this happens when you make two requests at the same time for the same static resource

**Comment by haijian-vaadin** [comment]
Seems this makes the Spring build unstable, e.g. this [build failure](https://bender.vaadin.com/buildConfiguration/Flow_Flow80_AddOns_Spring170Snapshot/234031)

**Comment by haijian-vaadin** [comment]
Also my [Spring PR](https://github.com/vaadin/spring/pull/812) is constantly failing because of this issue.

**Comment by estronque** [comment]
It seems this issue is also related to [Issue](https://github.com/vaadin/flow/issues/11230). An exception is raised in the method 'resourceIsDirectory' for resources with protocol "jar:war:file"

**Comment by jflamy** [comment]
Confirming that #11378 occurs in 14.6.5; perhaps the fix to this bug can be reviewed to cover #11378 as well.
